lib/datadog/core/telemetry/v1/dependency.rb

```diff
@@ -17,9 +17,10 @@ class Dependency
 
           # @param name [String] Module name
           # @param version [String] Version of resolved module
-          # @param hash [String] Dependency hash
+          # @param hash [String] Dependency hash, in case `version` is not available
           def initialize(name:, version: nil, hash: nil)
             raise ArgumentError, ERROR_NIL_NAME_MESSAGE if name.nil?
+            raise ArgumentError, 'if both :version and :hash exist, use :version only' if version && hash
```

Comment:
I soft-wonder if we should just remove the support for the `hash` entirely, since it would never be applicable to Ruby, right?

Reply:
I'm not so sure it won't be.

But I think at the end of day, we'll probably move to `telemetry/v2` altogether before (which changes most models), before this could become an issue.